hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/ExpressionEvalContext.java

```diff
@@ -67,8 +74,29 @@ static ExpressionEvalContext from(Context ctx) {
             return new ExpressionEvalContextImpl(
                     arguments,
                     new DefaultSerializationServiceBuilder().build(),
-                    Util.getNodeEngine(ctx.hazelcastInstance()));
+                    Util.getNodeEngine(ctx.hazelcastInstance())
+            );
+        }
+    }
+
+    static ExpressionEvalContext createContext(
+            @Nonnull List<Object> arguments,
+            @Nonnull NodeEngine nodeEngine,
+            @Nonnull InternalSerializationService iss,
+            @Nullable SqlSecurityContext ssc) {
+        ExpressionEvalContext newEec = new ExpressionEvalContextImpl(arguments, iss, nodeEngine);
+        if (ssc != null) {
+            newEec.setSecurityContext(ssc);
```

Comment:
IMO too many null checks. I do not have anything against using `NoOpSqlSecurityContext`. With nulls we have more a complicated strategy design pattern, where apart from real strategies we also have null strategy.

Reply:
see `SqlExecuteMessageTask.prepareSecurityContext()`

Reply:
let's have a quick chat about it